controllers/DownloadController.js

```diff
@@ -50,7 +50,7 @@ class DownloadController {
         res.header('Cache-Control', 'private, no-cache, no-store, must-revalidate');
         res.header('Expires', '-1');
         res.header('Pragma', 'no-cache');
-      if(!isNaN(parseFloat(req.query.bufferSize)) && (parseFloat(req.query.bufferSize)<=532421875)){
+      if(!isNaN(parseInt(req.query.bufferSize)) && (isFinite(req.query.bufferSize)) && (parseInt(req.query.bufferSize)<=532421875)){
```

Comment:
Can we make sure it's not negative? Instead of hardcoding the 500 MB limit, is it possible to have it as a configurable parameter (with the default value of 500 MB)? Depending on how people are using the library, different max values may be needed. If we make it a config parameter and validate it when reading it in, we don't need the `isFinite` as we are requiring buffers of size up to that parameter value.